PSI-PROF initial submission to round 19

jturtle opened the pull request:
The trajectories are either in, or very close to, their final state.  So I wanted to start the validation process to ensure that our format is compliant.

We will be updating the model metadata to a more complete form later today.

Trajectories are paired across horizon, age, scenario, and target.

LucieContamin commented:
Hi @shauntruelove , 
Thanks for the pull-request. Could I ask if the pull-request is ready to be merged or do you prefer to keep it open for more update?
Many thanks,
Lucie

jturtle committed:
PSI-PROF initial submission to round 19

LucieContamin commented:
Hi @jturtle ,

Thank you for the files submissions and the information. 
Please let me know if any issues or questions!

Best, Lucie

jturtle committed:
Update submission and metadata formatting

jturtle committed:
update metadata formatting - take 2

LucieContamin reviewed (changes requested):
Hi @jturtle ,

Thanks for the update. the metadata file format is ok now, but the model output file still need some update. It seems that the validation fails because of the format of some columns, especially the `location`. Would it be possible to change it as character? It should solve the first issue. 

I will work on the validation to have a clearer output. Please let me know if any questions or issues

Thanks, Lucie

jturtle committed:
Update submission format

Thank you for the feedback.  I had three columns in R's 'factor' format.  These have been reverted to character types.

LucieContamin reviewed (approved):
Thank you very much for the update. I can ignore the "flat" projection warning if it seems expected to you, and merge the PR!

Thanks again

jturtle commented:
DC is quite small.  I think a flat "0" trajectory for mortality is definitely possible.  I will double-check the results later today, but I think it is okay to merge now.  Thanks!

jturtle commented:
Hi Lucie,
I double-checked one of the constant-value trajectories listed in the validations, and I do not get constant values on my end (see attached screenshot).
![Screenshot 2025-05-15 at 4 31 19 PM](https://github.com/user-attachments/assets/304ede44-2133-4de0-ad76-7db2150be0d3)

Specifically I am checking against this validation output: "C-2025-04-01, 0-64, 11, 2025-04-27, inc death, sample, 7217". Here I was assuming that 7217 is the run_grouping, and 11 is the location FIPS.  But my 7217 run_grouping is associated with FIPS location "28", so the 11 must mean something else?

If I assume that the "11" is FIPS for DC, and search all trajectories, I do have some mortality 0-64 trajectories that never exceed 0.5.  However I am submitting different values between 0.0 and 0.4.  Are you rounding to whole numbers before running the flat trajectory check?

Thanks, Jamie

LucieContamin commented:
Hi Jamie,

The sample id number is misleading, I apologize about that. It's created by concatenation of the `run_grouping` and `stochastic_run` column. I will add more detail in the validation documentation. 

So, to replicate:

```r
df <- arrow::read_parquet("model-output/PSI-PROF/2025-04-27-PSI-PROF.gz.parquet")
df$output_type_id <- as.numeric(as.factor(paste0(df$run_grouping, "-", df$stochastic_run)))
df |> dplyr::filter(output_type_id == 7217, age_group == "0-64", target == "inc death", scenario_id == "C-2025-04-01", location == "11") 

#># A tibble: 52 × 11
#>   scenario_id  age_group location horizon run_grouping value origin_date target    output_type output_type_id #>stochastic_run
#>   <chr>        <chr>     <chr>      <int>        <dbl> <dbl> <chr>       <chr>     <chr>                <dbl>          <dbl>
#> 1 C-2025-04-01 0-64      11             1         2452   0.1 2025-04-27  inc death sample                7217           2452
#> 2 C-2025-04-01 0-64      11             2         2452   0.1 2025-04-27  inc death sample                7217           2452
#> 3 C-2025-04-01 0-64      11             3         2452   0.1 2025-04-27  inc death sample                7217           2452
#> 4 C-2025-04-01 0-64      11             4         2452   0.1 2025-04-27  inc death sample                7217           2452
#> 5 C-2025-04-01 0-64      11             5         2452   0.1 2025-04-27  inc death sample                7217           2452
#> 6 C-2025-04-01 0-64      11             6         2452   0.1 2025-04-27  inc death sample                7217           2452
#> 7 C-2025-04-01 0-64      11             7         2452   0.1 2025-04-27  inc death sample                7217           2452
#> 8 C-2025-04-01 0-64      11             8         2452   0.1 2025-04-27  inc death sample                7217           2452
#> 9 C-2025-04-01 0-64      11             9         2452   0.1 2025-04-27  inc death sample                7217           2452
#>10 C-2025-04-01 0-64      11            10         2452   0.1 2025-04-27  inc death sample                7217           2452
#># ℹ 42 more rows
#># ℹ Use `print(n = ...)` to see more rows

test <- dplyr::filter(df, output_type_id == 7217, age_group == "0-64", target == "inc death", scenario_id == "C-2025-04-01", location == "11") 
unique(test$value)

#>[1] 0.1

```
